From fb59a892c4a2b900b929acbf05578d831749e3ba Mon Sep 17 00:00:00 2001
From: "Haelwenn (lanodan) Monnier" <contact@hacktivis.me>
Date: Thu, 11 Apr 2024 15:54:07 +0200
Subject: [PATCH] rsvg-css: Fix xmlError function pointer type

---
 rsvg-css.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/rsvg-css.c b/rsvg-css.c
index 5b359a20..95073cf2 100644
--- a/rsvg-css.c
+++ b/rsvg-css.c
@@ -839,7 +839,7 @@ rsvg_css_parse_overflow (const char *str, gboolean * inherit)
 }
 
 static void
-rsvg_xml_noerror (void *data, xmlErrorPtr error)
+rsvg_xml_noerror (void *data, const struct _xmlError *error)
 {
 }
 
@@ -863,7 +863,7 @@ rsvg_css_parse_xml_attribute_string (const char *attribute_string)
 
     memset (&handler, 0, sizeof (handler));
     xmlSAX2InitDefaultSAXHandler (&handler, 0);
-    handler.serror = rsvg_xml_noerror;
+    handler.serror = &rsvg_xml_noerror;
     parser = xmlCreatePushParserCtxt (&handler, NULL, tag, strlen (tag) + 1, NULL);
     parser->options |= XML_PARSE_NONET;
 
-- 
2.43.2

