$OpenBSD: patch-Makefile_in,v 1.4 2013/03/12 00:59:50 brad Exp $
--- portaudio/Makefile.in.orig	Thu Jul  5 23:44:34 2012
+++ portaudio/Makefile.in	Thu Jul  5 23:49:34 2012
@@ -146,6 +146,7 @@ SRC_DIRS = \
 	src/hostapi/dsound \
 	src/hostapi/jack \
 	src/hostapi/oss \
+	src/hostapi/sndio \
 	src/hostapi/wasapi \
 	src/hostapi/wdmks \
 	src/hostapi/wmme \
$OpenBSD: patch-configure_in,v 1.5 2019/02/06 14:21:15 jca Exp $
Index: configure.in
--- portaudio/configure.in.orig
+++ portaudio/configure.in
@@ -24,6 +24,10 @@ AC_ARG_WITH(alsa,
             AS_HELP_STRING([--with-alsa], [Enable support for ALSA @<:@autodetect@:>@]),
             [with_alsa=$withval])
 
+AC_ARG_WITH(sndio,
+            AS_HELP_STRING([--with-sndio], [Enable support for sndio @<:@autodetect@:>@]),
+            [with_sndio=$withval])
+
 AC_ARG_WITH(jack,
             AS_HELP_STRING([--with-jack], [Enable support for JACK @<:@autodetect@:>@]),
             [with_jack=$withval])
@@ -120,6 +124,10 @@ have_alsa=no
 if test "x$with_alsa" != "xno"; then
     AC_CHECK_LIB(asound, snd_pcm_open, have_alsa=yes, have_alsa=no)
 fi
+have_sndio=no
+if test "x$with_sndio" != "xno"; then
+    AC_CHECK_LIB(sndio, sio_open, have_sndio=yes, have_sndio=no)
+fi
 have_asihpi=no
 if test "x$with_asihpi" != "xno"; then
     AC_CHECK_LIB(hpi, HPI_SubSysCreate, have_asihpi=yes, have_asihpi=no, -lm)
@@ -406,6 +414,13 @@ case "${host_os}" in
            AC_DEFINE(PA_USE_ALSA,1)
         fi
 
+        if [[ "$have_sndio" = "yes" -a "$with_sndio" != "no" ]] ; then
+           DLL_LIBS="$DLL_LIBS -lsndio"
+           LIBS="$LIBS -lsndio"
+           OTHER_OBJS="$OTHER_OBJS src/hostapi/sndio/pa_sndio.o"
+           AC_DEFINE(PA_USE_SNDIO,1)
+        fi
+
         if [[ "$have_jack" = "yes" ] && [ "$with_jack" != "no" ]] ; then
            DLL_LIBS="$DLL_LIBS $JACK_LIBS"
            CFLAGS="$CFLAGS $JACK_CFLAGS"
@@ -509,6 +524,7 @@ case "$target_os" in
         ;;
      *)
 	AC_MSG_RESULT([
+  Sndio ....................... $have_sndio
   OSS ......................... $have_oss
   JACK ........................ $have_jack
 ])
$OpenBSD: patch-include_portaudio_h,v 1.3 2019/02/06 14:21:15 jca Exp $
Index: include/portaudio.h
--- portaudio/include/portaudio.h.orig
+++ portaudio/include/portaudio.h
@@ -287,7 +287,8 @@ typedef enum PaHostApiTypeId
     paWDMKS=11,
     paJACK=12,
     paWASAPI=13,
-    paAudioScienceHPI=14
+    paAudioScienceHPI=14,
+    paSndio=15
 } PaHostApiTypeId;
 
 
$OpenBSD: patch-src_os_unix_pa_unix_hostapis_c,v 1.2 2013/03/12 00:59:50 brad Exp $
--- portaudio/src/os/unix/pa_unix_hostapis.c.orig	Thu Jul  5 23:44:45 2012
+++ portaudio/src/os/unix/pa_unix_hostapis.c	Thu Jul  5 23:49:33 2012
@@ -44,6 +44,7 @@
 
 PaError PaJack_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex index );
 PaError PaAlsa_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex index );
+PaError PaSndio_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex index );
 PaError PaOSS_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex index );
 /* Added for IRIX, Pieter, oct 2, 2003: */
 PaError PaSGI_Initialize( PaUtilHostApiRepresentation **hostApi, PaHostApiIndex index );
@@ -78,6 +79,10 @@ PaUtilHostApiInitializer *paHostApiInitializers[] =
 #endif
 
 #endif  /* __linux__ */
+
+#ifdef PA_USE_SNDIO
+	PaSndio_Initialize,
+#endif
 
 #if PA_USE_JACK
         PaJack_Initialize,
