From 21146f87333d9d9c1f3ec1e1e4c2e311193d388f Mon Sep 17 00:00:00 2001
From: Haelwenn Monnier <lanodan@users.noreply.github.com>
Date: Sat, 11 Nov 2017 15:22:20 +0100
Subject: [PATCH] Remove --autoplay and --pause short options

From oshu.c it looks like there is no short options for ``--autoplay`` and ``--pause``.
---
 docs/oshu.1.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/docs/oshu.1.in b/docs/oshu.1.in
index e8b7f3e..b24ef66 100644
--- a/docs/oshu.1.in
+++ b/docs/oshu.1.in
@@ -29,10 +29,10 @@ twice and you'll get verbose debugging messages.
 \fB\-h, \-\-help\fR
 Show a brief help message.
 .TP
-\fB\-h, \-\-autoplay\fR
+\fB\-\-autoplay\fR
 Perform a perfect run on the beatmap without any user interaction.
 .TP
-\fB\-h, \-\-pause\fR
+\fB\-\-pause\fR
 Start the game in a paused state. Might be useful when you're starting the game
 from a terminal as you won't be holding your mouse when the game starts.
 
