diff --git a/pm_common/CMakeLists.txt b/pm_common/CMakeLists.txt
index e171047..2f191f9 100644
--- a/pm_common/CMakeLists.txt
+++ b/pm_common/CMakeLists.txt
@@ -78,9 +78,11 @@ if(UNIX)
 
     set(LINUXSRC pmlinuxalsa pmlinux finddefault)
     prepend_path(LIBSRC ../pm_linux/ ${LINUXSRC})
+    set(SNDIOSRC pmsndio)
+    prepend_path(LIBSRC ../pm_sndio/ ${SNDIOSRC})
     list(APPEND LIBSRC ../porttime/ptlinux)
 
-    set(PM_NEEDED_LIBS pthread asound)
+    set(PM_NEEDED_LIBS pthread asound sndio)
   endif(APPLE)
 else(UNIX)
   if(WIN32)
diff --git a/pm_dylib/CMakeLists.txt b/pm_dylib/CMakeLists.txt
index f693dd6..aa57173 100644
--- a/pm_dylib/CMakeLists.txt
+++ b/pm_dylib/CMakeLists.txt
@@ -83,9 +83,11 @@ if(UNIX)
 
     set(LINUXSRC pmlinuxalsa pmlinux finddefault)
     prepend_path(LIBSRC ../pm_linux/ ${LINUXSRC})
+    set(SNDIOSRC pmsndio)
+    prepend_path(LIBSRC ../pm_sndio/ ${SNDIOSRC})
     list(APPEND LIBSRC ../porttime/ptlinux)
 
-    set(PM_NEEDED_LIBS pthread asound)
+    set(PM_NEEDED_LIBS pthread asound sndio)
   endif(APPLE)
 else(UNIX)
   if(WIN32)
