From efb2d5ef9fb64d681036a617c40e42be29ab3e9c Mon Sep 17 00:00:00 2001
From: "Haelwenn (lanodan) Monnier" <contact@hacktivis.me>
Date: Fri, 14 Jun 2024 13:13:14 +0200
Subject: [PATCH] scons: $AR, disable stripping

---
 SConstruct | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/SConstruct b/SConstruct
index 8d2d0b4eb8..b580977bac 100644
--- a/SConstruct
+++ b/SConstruct
@@ -290,6 +290,7 @@ opts.Add(BoolVariable("builtin_zstd", "Use the built-in Zstd library", True))
 # be overridden for a specific platform), the lowercase ones are appended.
 opts.Add("CXX", "C++ compiler binary")
 opts.Add("CC", "C compiler binary")
+opts.Add("AR", "Archiver binary")
 opts.Add("LINK", "Linker binary")
 opts.Add("cppdefines", "Custom defines for the pre-processor")
 opts.Add("ccflags", "Custom flags for both the C and C++ compilers")
@@ -751,8 +752,6 @@ else:
         if methods.is_apple_clang(env):
             # Apple Clang, its linker doesn't like -s.
             env.Append(LINKFLAGS=["-Wl,-S", "-Wl,-x", "-Wl,-dead_strip"])
-        else:
-            env.Append(LINKFLAGS=["-s"])
 
     # Linker needs optimization flags too, at least for Emscripten.
     # For other toolchains, this _may_ be useful for LTO too to disambiguate.
-- 
2.45.2

