commit 620a70311df192d52957d635a9f2753b265ff61e
Author: rakshasa <sundell.software@gmail.com>
Date:   Thu Mar 10 02:44:24 2016 +0900

    Fixed minor C++ standard compliance issue.

diff --git a/src/dht/dht_transaction.cc b/src/dht/dht_transaction.cc
index f876e70d..84ffc175 100644
--- a/src/dht/dht_transaction.cc
+++ b/src/dht/dht_transaction.cc
@@ -46,14 +46,15 @@
 namespace torrent {
 
 DhtSearch::DhtSearch(const HashString& target, const DhtBucket& contacts)
-  : base_type(dht_compare_closer(m_target = target)),
+  : base_type(dht_compare_closer(target)),
     m_pending(0),
     m_contacted(0),
     m_replied(0),
     m_concurrency(3),
     m_restart(false),
     m_started(false),
-    m_next(end()) {
+    m_next(end()),
+    m_target(target) {
 
   add_contacts(contacts);
 }
