diff --git a/CMakeLists.txt.old b/CMakeLists.txt
index 771a52b..50c5cb9 100644
--- a/CMakeLists.txt.old
+++ b/CMakeLists.txt
@@ -259,8 +259,8 @@ IF(FREEGLUT_GLES)
       find_library(OPENGL_glu_LIBRARY NAME GLU HINTS ${X11_LIB_PATH})
     endif()
 
-    FIND_PACKAGE(OpenGL REQUIRED)
-    LIST(APPEND LIBS ${OPENGL_gl_LIBRARY})
+    FIND_PACKAGE(OpenGL COMPONENTS OpenGL REQUIRED)
+    LIST(APPEND LIBS ${OPENGL_opengl_LIBRARY})
     INCLUDE_DIRECTORIES(${OPENGL_INCLUDE_DIR})
 ENDIF()
 
