commit a3e5824c91a1b65b658c6a879df22e02e5d5815c
Author: Haelwenn (lanodan) Monnier <contact@hacktivis.me>
Date:   2019-07-28T22:10:25 GMT

    gdbus-codegen external

diff --git a/docs/reference/gio/meson.build b/docs/reference/gio/meson.build
index 930a3b7..d893170 100644
--- a/docs/reference/gio/meson.build
+++ b/docs/reference/gio/meson.build
@@ -164,7 +164,6 @@ if get_option('gtk_doc')
       'gsettings.xml',
       'gresource.xml',
       'gdbus.xml',
-      'gdbus-codegen.xml',
     ],
     expand_content_files : [
       'overview.xml',
@@ -172,7 +171,6 @@ if get_option('gtk_doc')
       'migrating-gnome-vfs.xml',
       'migrating-gconf.xml',
       'migrating-gdbus.xml',
-      'gdbus-codegen.xml',
     ],
     html_assets : [
       'gvfs-overview.png',
@@ -192,7 +190,7 @@ endif
 if get_option('man')
   manpages = ['gapplication', 'gio-querymodules', 'glib-compile-schemas',
               'glib-compile-resources', 'gsettings', 'gresource', 'gdbus',
-              'gio', 'gdbus-codegen']
+              'gio']
   foreach page : manpages
     custom_target(page + '-man',
       input: page + '.xml',
diff --git a/gio/meson.build b/gio/meson.build
index f0e08b4..d39ff60 100644
--- a/gio/meson.build
+++ b/gio/meson.build
@@ -232,7 +232,7 @@ gdbus_sources = files(
 )
 
 # Generate gdbus-codegen
-subdir('gdbus-2.0/codegen')
+#subdir('gdbus-2.0/codegen')
 
 # Generate xdp-dbus.{c,h}
 xdp_dbus_generated = custom_target('xdp-dbus',
@@ -241,8 +241,7 @@ xdp_dbus_generated = custom_target('xdp-dbus',
              'org.freedesktop.portal.ProxyResolver.xml',
              'org.freedesktop.portal.Trash.xml'],
     output : ['xdp-dbus.h', 'xdp-dbus.c'],
-    depend_files : gdbus_codegen_built_files,
-    command : [python, gdbus_codegen,
+    command : [find_program('gdbus-codegen'),
                '--interface-prefix', 'org.freedesktop.portal.',
                '--output-directory', '@OUTDIR@',
                '--generate-c-code', 'xdp-dbus',
@@ -263,8 +262,7 @@ xdp_dbus_generated = custom_target('xdp-dbus',
 gdbus_daemon_generated = custom_target('gdbus-daemon-generated',
     input : ['dbus-daemon.xml'],
     output : ['gdbus-daemon-generated.h', 'gdbus-daemon-generated.c'],
-    depend_files : gdbus_codegen_built_files,
-    command : [python, gdbus_codegen,
+    command : [find_program('gdbus-codegen'),
                '--interface-prefix', 'org.',
                '--output-directory', '@OUTDIR@',
                '--generate-c-code', 'gdbus-daemon-generated',
diff --git a/gio/tests/gdbus-object-manager-example/meson.build b/gio/tests/gdbus-object-manager-example/meson.build
index 404c377..50cf2d8 100644
--- a/gio/tests/gdbus-object-manager-example/meson.build
+++ b/gio/tests/gdbus-object-manager-example/meson.build
@@ -5,7 +5,7 @@ gdbus_example_objectmanager_generated = custom_target('objectmanager-gen',
             'objectmanager-gen.c',
             'objectmanager-gen-org.gtk.GDBus.Example.ObjectManager.Animal.xml',
             'objectmanager-gen-org.gtk.GDBus.Example.ObjectManager.Cat.xml'],
-  command : [python, gdbus_codegen,
+  command : [find_program('gdbus-codegen'),
              '--interface-prefix', 'org.gtk.GDBus.Example.ObjectManager.',
              '--c-namespace', 'Example',
              '--c-generate-object-manager',
diff --git a/gio/tests/meson.build b/gio/tests/meson.build
index a0c0944..9025c41 100644
--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -211,8 +211,7 @@ if host_machine.system() != 'windows'
         input :   ['test-codegen.xml'],
         output :  ['gdbus-test-codegen-generated.h',
                    'gdbus-test-codegen-generated.c'],
-        depend_files : gdbus_codegen_built_files,
-        command : [python, gdbus_codegen,
+        command : [find_program('gdbus-codegen'),
                    '--interface-prefix', 'org.project.',
                    '--output-directory', '@OUTDIR@',
                    '--generate-c-code', 'gdbus-test-codegen-generated',
@@ -227,8 +226,7 @@ if host_machine.system() != 'windows'
       custom_target('gdbus-test-codegen-generated-interface-info-h',
           input :   ['test-codegen.xml'],
           output :  ['gdbus-test-codegen-generated-interface-info.h'],
-          depend_files : gdbus_codegen_built_files,
-          command : [python, gdbus_codegen,
+          command : [find_program('gdbus-codegen'),
                      '--interface-info-header',
                      annotate_args,
                      '--output', '@OUTPUT@',
@@ -236,8 +234,7 @@ if host_machine.system() != 'windows'
       custom_target('gdbus-test-codegen-generated-interface-info-c',
           input :   ['test-codegen.xml'],
           output :  ['gdbus-test-codegen-generated-interface-info.c'],
-          depend_files : gdbus_codegen_built_files,
-          command : [python, gdbus_codegen,
+          command : [find_program('gdbus-codegen'),
                      '--interface-info-body',
                      annotate_args,
                      '--output', '@OUTPUT@',
