--- oss-v4.2-build2019-src-gpl/setup/Linux/build.sh.old	2020-04-22 23:19:18.042367733 +0200
+++ oss-v4.2-build2019-src-gpl/setup/Linux/build.sh	2020-04-22 23:20:27.526622099 +0200
@@ -132,18 +132,18 @@
 	if test -f $SRCDIR/kernel/drv/$N/$N.man
 	then
 	  sed "s:CONFIGFILEPATH:$OSSLIBDIR/conf:g" < $SRCDIR/kernel/drv/$N/$N.man > /tmp/ossman.txt
-	  $TXT2MAN -t "$CMD" -v "OSS Devices" -s 7 /tmp/ossman.txt | gzip -9 > prototype/usr/share/man/man7/$N.7.gz
+	  $TXT2MAN -t "$CMD" -v "OSS Devices" -s 7 /tmp/ossman.txt > prototype/usr/share/man/man7/$N.7
 	else
 		if test -f $SRCDIR/kernel/nonfree/drv/$N/$N.man
 		then
 	  		sed "s:CONFIGFILEPATH:$OSSLIBDIR/conf:g" < $SRCDIR/kernel/nonfree/drv/$N/$N.man > /tmp/ossman.txt
-	  		$TXT2MAN -t "$CMD" -v "OSS Devices" -s 7 $SRCDIR/kernel/nonfree/drv/$N/$N.man | gzip -9 > prototype/usr/share/man/man7/$N.7.gz
+	  		$TXT2MAN -t "$CMD" -v "OSS Devices" -s 7 $SRCDIR/kernel/nonfree/drv/$N/$N.man > prototype/usr/share/man/man7/$N.7
 		fi
 	fi
 done
 
 sed "s:CONFIGFILEPATH:$OSSLIBDIR/conf:g" < $SRCDIR/kernel/drv/osscore/osscore.man > /tmp/ossman.txt
-$TXT2MAN -t "osscore" -v "OSS Devices" -s 7 /tmp/ossman.txt | gzip -9 > prototype/usr/share/man/man7/osscore.7.gz
+$TXT2MAN -t "osscore" -v "OSS Devices" -s 7 /tmp/ossman.txt > prototype/usr/share/man/man7/osscore.7
 rm -f /tmp/ossman.txt
 
 # Link the optional NOREGPARM modules
@@ -162,13 +162,13 @@
 do
 	N=`basename $n .man`
 
-	$TXT2MAN -t "$CMD" -v "OSS Devices" -s 7 $n | gzip -9 > prototype/usr/share/man/man7/$N.7.gz
+	$TXT2MAN -t "$CMD" -v "OSS Devices" -s 7 $n > prototype/usr/share/man/man7/$N.7
 done
 
 for n in $SRCDIR/misc/man1m/*.man
 do
 	N=`basename $n .man`
-	$TXT2MAN -t "$CMD" -v "OSS System Administration Commands" -s 1 $n | gzip -9 > prototype/usr/share/man/man1/$N.1.gz
+	$TXT2MAN -t "$CMD" -v "OSS System Administration Commands" -s 1 $n > prototype/usr/share/man/man1/$N.1
 done
 
 if ! cp lib/libOSSlib/libOSSlib.so lib/libsalsa/.libs/libsalsa.so.2.0.0 prototype/$OSSLIBDIR/lib
@@ -189,7 +189,7 @@
 for i in target/bin/*
 do
 CMD=`basename $i`
-$TXT2MAN -t "$CMD" -v "OSS User Commands" -s 1 cmd/$CMD/$CMD.man | gzip -9 > prototype/usr/share/man/man1/$CMD.1.gz
+$TXT2MAN -t "$CMD" -v "OSS User Commands" -s 1 cmd/$CMD/$CMD.man > prototype/usr/share/man/man1/$CMD.1
 echo done $CMD
 done
 
@@ -198,12 +198,12 @@
   CMD=`basename $i`
   if test -f cmd/$CMD/$CMD.man
   then
-	$TXT2MAN -t "$CMD" -v "OSS System Administration Commands" -s 8 cmd/$CMD/$CMD.man | gzip -9 > prototype/usr/share/man/man8/$CMD.8.gz
+	$TXT2MAN -t "$CMD" -v "OSS System Administration Commands" -s 8 cmd/$CMD/$CMD.man > prototype/usr/share/man/man8/$CMD.8
 	echo done $CMD
   fi
 done
 
-$TXT2MAN -t "ossdetect" -v "User Commands" -s 8 os_cmd/Linux/ossdetect/ossdetect.man | gzip -9 > prototype/usr/share/man/man8/ossdetect.8.gz
+$TXT2MAN -t "ossdetect" -v "User Commands" -s 8 os_cmd/Linux/ossdetect/ossdetect.man > prototype/usr/share/man/man8/ossdetect.8
 echo done ossdetect
 
 cp -f $SRCDIR/oss/lib/flashsupport.c prototype/$OSSLIBDIR/lib
