--- src/gif.c	2017-03-19 18:53:14.440665470 +0100
+++ src/gif.c	2017-03-19 18:53:18.856664983 +0100
@@ -32,7 +32,11 @@
 	/* HACK HACK HACK */
 	rewind(f);
 	lseek(fileno(f), 0L, SEEK_SET);
-	if(!(gif = DGifOpenFileHandle(fileno(f)))){
+#if defined(GIFLIB_MAJOR) && GIFLIB_MAJOR >= 5
+	if(!(gif = DGifOpenFileHandle(fileno(f), NULL))){
+#else
+	if(!(gif = DGifOpenFileHandle(fileno(f)))):
+#endif
 		/* HACK AND HOPE */
 		rewind(f);
 		lseek(fileno(f), 0L, SEEK_SET);
