diff --git a/nnrpd/tls.c b/nnrpd/tls.c
index 97609c2..1a3dba7 100644
--- a/nnrpd/tls.c
+++ b/nnrpd/tls.c
@@ -501,7 +501,7 @@ tls_init_serverengine(int verifydepth, int askcert, int requirecert,
       syslog(L_NOTICE, "starting TLS engine");

 /* New functions have been introduced in OpenSSL 1.1.0. */
-#if OPENSSL_VERSION_NUMBER < 0x010100000L
+#if OPENSSL_VERSION_NUMBER < 0x010100000L || defined(LIBRESSL_VERSION_NUMBER)
     SSL_load_error_strings();
     SSLeay_add_ssl_algorithms();
     CTX = SSL_CTX_new(SSLv23_server_method());
