From beb9839495a282414dab3dbe36ea39075e2a979e Mon Sep 17 00:00:00 2001
From: Christopher O'Neill <code@chrisoneill.co.uk>
Date: Wed, 28 Feb 2018 19:35:18 +0000
Subject: [PATCH] Fix #152: Type mismatch in EQConstants.cpp

---
 src/tracker/EQConstants.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tracker/EQConstants.cpp b/src/tracker/EQConstants.cpp
index ffbf6ea8..f221364c 100644
--- a/src/tracker/EQConstants.cpp
+++ b/src/tracker/EQConstants.cpp
@@ -23,7 +23,7 @@
 #include "EQConstants.h"
 #include "math.h"
 
-#define p(e) ldexp(0x10558, e-4*4)
+#define p(e) ldexp(66904.0f, e-4*4)
 
 const float EQConstants::EQ3bands[3] = 
 {
