diff --git a/cmake/FindSDL2.cmake.old b/cmake/FindSDL2.cmake
index 27c78f0..0506fc4 100644
--- a/cmake/FindSDL2.cmake.old
+++ b/cmake/FindSDL2.cmake
@@ -93,7 +93,7 @@ if(SDL2_INCLUDE_DIR AND EXISTS "${SDL2_INCLUDE_DIR}/SDL_version.h")
 endif()
 
 set(SDL2_INCLUDE_DIRS ${SDL2_INCLUDE_DIR})
-set(SDL2_LIBRARIES ${SDL2MAIN_LIBRARY} ${SDL2_LIBRARY})
+set(SDL2_LIBRARIES ${SDL2_LIBRARY})
 
 include(FindPackageHandleStandardArgs)
 
